cli/githubutil.py

```diff
@@ -71,6 +85,213 @@ def assign_github_team_to_repo(
         team_name=team_name
     )
 
+def generate_release_notes_cli(
+    repo_dir: str,
+    github_cfg_name: str,
+    github_repository_owner: str,
+    github_repository_name: str,
+    repository_branch: str,
+    commit_range: str=None
+):
+    github_cfg = ctx().cfg_factory().github(github_cfg_name)
+    helper = GitHubRepositoryHelper(
+        github_cfg=github_cfg,
+        owner=github_repository_owner,
+        name=github_repository_name,
+        default_branch=repository_branch,
+    )
+    generate_release_notes(repo_dir=repo_dir, helper=helper, repository_branch=repository_branch, commit_range=commit_range)
+
+def generate_release_notes(
+    repo_dir: str,
+    helper: GitHubRepositoryHelper,
+    repository_branch: str,
+    commit_range: str=None
+):
+    repo = git.Repo(existing_dir(repo_dir))
+    if repo.active_branch.name != repository_branch:
+        fail('need to switch to branch {branch}. Current active branch: {active_branch}'.format(branch=repository_branch, active_branch=repo.active_branch.name))
+
+    if not commit_range:
+        commit_range = calculate_range(repo, helper)
+    pr_numbers = fetch_pr_numbers_in_range(repo, commit_range)
+    release_note_objs = fetch_release_notes_from_prs(helper, pr_numbers)
+    release_notes = build_markdown(release_note_objs)
```

Comment:
rename to release_notes_str or sth similar to express the difference between the objects and the resulting release-note-"text"